math/src/main/java/smile/math/matrix/Matrix.java

```diff
@@ -456,6 +456,32 @@ default SVD svd(int k, double kappa, int maxIter) {
             return new SVD(U, V, s);
         }
     }
+    
+    /**
+     * In linear algebra, a Toeplitz matrix or diagonal-constant matrix, in
+     * which each descending diagonal from left to right is constant.
+     * 
+     * @param input
+     *            original data to construct the toeplitz matrix
+     * @return a {@link DenseMatrix} of Toplietz characteristic
+     */
+    static DenseMatrix toeplitz(double[] input) {
+        int length = input.length;
+        double toeplitz[][] = new double[length][length];
+
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < length; j++) {
+                if (j > i) {
+                    toeplitz[i][j] = input[j - i];
```

Comment:
So this is also a symmetric matrix?

Reply:
I believe yes, it is symmetric. https://en.wikipedia.org/wiki/Toeplitz_matrix